src/utils.rs

```diff
@@ -10,6 +10,69 @@
 
 use syntax::ast::Visibility;
 
+pub trait FindUncommented {
+    fn find_uncommented(&self, pat: &str) -> Option<usize>;
+}
+
+impl FindUncommented for str {
+    fn find_uncommented(&self, pat: &str) -> Option<usize> {
+        let needle = pat.as_bytes();
+        let needle_len = needle.len();
+        let mut match_count: usize = 0;
+        let mut possible_comment = false;
+
+        for (i, b) in self.as_bytes().iter().cloned().enumerate() {
+            if b == needle[match_count] {
+                match_count += 1;
+                if match_count == needle_len {
+                    return Some((i + 1) - needle_len);
+                }
+            } else {
+                match_count = 0;
+            }
+
+            if possible_comment {
+                if b == '/' as u8 {
```

Comment:
Ah, cool, I'm glad you thought about it. Could you leave a comment to this effect please? I do fear a little bit that this is premature optimisation - the char should be on the stack so no allocation and there will be copies from memory to registers any way, so hopefully, post-optimisation, the code should be similar. If the overhead for checking the char width is significant, then this would be faster - I wouldn't put money either way without measuring. Anyway, just rambling - I'm happy to leave it with a comment.